### DISCUSSION TITLE
Implement QR code scanning

### DISCUSSION THREAD

**Discussion by tavurth** [godot-extended-libraries/godot-ideas#39] [Ideas]
### Describe the project you are working on

Any project where the developer would like to use a QR code for sharing data

### Describe the problem or limitation you are having in your project

I would like to implement a QR code scanner, but currently Godot doesn't really have something of this nature.

I implemented a simple camera here:

https://github.com/tavurth/godot-camera

However to get the QR code scanner working I'll probably have to leverage some C++ library (openCV)

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

A module or plugin which can quickly detect a QR code from a passed `ImageTexture2D`

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

`ARServer.scan_for_qr(ImageTexture2D)`

### If this enhancement will not be used often, can it be worked around with a few lines of script?

I feel like this would be quite a useful implementation for the core as it's something which is very prevalent in society and yet currently it's rather difficult to implement.

### Is there a reason why this should be core and not an add-on in the asset library?

Having this as a plugin would also be useful, and I am noting it here to get it on the radar.

https://github.com/godot-extended-libraries/godot-ideas/issues/38#event-5617601066

https://github.com/godotengine/godot-proposals/issues/3547

**Comment by Xrayez** [comment]
I converted this into a discussion, see https://github.com/godot-extended-libraries/godot-ideas/discussions/18.

Regarding the proposal, perhaps this is something which can be implemented in [Goost](https://goostengine.github.io/), but I'm afraid integrating libraries like OpenCV could be overkill for just QR code scanning.

I see there's a module https://github.com/williamd1k0/godot-qrcode, but looks like it's for generating QR codes, not scanning.

**Comment by tavurth** [comment]
Yea I agree, qr scanning shouldn't need opencv as most applications will use only the qr part from that library.

I found a comparison here

https://www.dynamsoft.com/codepool/qr-code-reading-benchmark-and-comparison.html

**Reply by tavurth** [reply]
http://zbar.sourceforge.net/

Zbar seems to be a fairly good option, LGPL licensed

**Reply by Calinou** [reply]
The LGPL makes it very difficult to use the library on iOS legally, so I wouldn't recommend it.

**Reply by tavurth** [reply]
Thank you! Good feedback 😊

**Comment by tavurth** [comment]
Unfortunately https://github.com/godotengine/godot/issues/46531 blocks this issue for me, as I wanted this on Android devices.

When 46531 is completed I've been building out a QR code scanner based on ZXing.Net using GodotMono which seems to work ok.

**Reply by tavurth** [reply]
https://github.com/tavurth/godot-qr-scanner

**Reply by tavurth** [reply]
Yes sure I agree it's helpful to have such a sample repo. Please feel free to contribute that!
